packages/components/src/Link/Link.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { forwardRef } from 'react'
+import { Link as RACLink, type LinkProps as RACLinkProps } from 'react-aria-components'
+import { type TextProps } from '~components/Text'
+import { mergeClassNames } from '~components/utils/mergeClassNames'
+import { LinkContent } from './subcomponents/LinkContent'
+import styles from './Link.module.css'
+
+export type LinkProps = (UnderlinedLink | NonUnderlinedLink) & (InlineLink | NonInlineLink)
+
+type BaseLinkProps = {
+  variant?: 'primary' | 'secondary' | 'white'
```

Comment:
Can we add some type descriptions for our types with the default values where relevant?

Doesn't have to be too long, but something like this will provide some good feedback via intellisense 😉 
```
  /** Controls the visual style of a link. @default 'primary' */
```